pkg/arvo/sys/vane/jael.hoon

```diff
@@ -1097,1181 +604,310 @@
+    |-  ^+  this-su
+    ?~  noy  this-su
     $(noy t.noy, moz [[i.noy cad] moz])
   ::
-  ++  vent-pass
-    |=  [yen=(set duct) res=vent-result]
+  ++  emit-peer
+    |=  [app=term =path]
+    %-  emit
+    :*  hen
+        %pass
+        [app path]
+        %g
+        %deal
+        [our our]
+        app
+        %peer
+        path
+    ==
+  ::
+  ++  peer
+    |=  [app=term whos=(set ship)]
+    ?:  =(~ whos)
+      (emit-peer app /)
+    =/  whol=(list ship)  ~(tap in whos)
+    |-  ^+  this-su
+    ?~  whol  this-su
+    =.  this-su  (emit-peer app /(scot %p i.whol))
+    $(whol t.whol)
+  ::
+  ++  public-keys-give
+    |=  [yen=(set duct) =public-keys-result]
     =+  yez=~(tap in yen)
-    |-  ^+  ..vent-pass
-    ?~  yez  ..vent-pass
-    =.  ..vent-pass
-      =*  d  i.yez
-      ?>  ?=([[%a @ @ *] *] d)
-      =+  our=(slav %p i.t.i.d)
-      =+  who=(slav %p i.t.t.i.d)
-      %+  exec  [d ~ ~]
-      :+  %pass
-        /(scot %p our)/vent-result
+    |-  ^+  this-su
+    ?~  yez  this-su
+    =*  d  i.yez
+    =.  this-su
+      ?.  &(?=([[%a @ @ *] *] d) !=(%pubs i.t.i.d))
+        %-  emit
+        [d %give %public-keys public-keys-result]
+      =/  our  (slav %p i.t.i.d)
+      =/  who  (slav %p i.t.t.i.d)
+      =/  =message  [%public-keys-result public-keys-result]
+      %-  emit
+      :^    d
+          %pass
+        /public-keys-result
       ^-  note
-      [%a %want who /j/(scot %p our)/vent-result %vent-result res]
+      [%a %want who /j/public-keys-result message]
     $(yez t.yez)
   ::
-  ++  extract-snap                                    ::  extract rewind point
-    ^-  snapshot
-    :*  kyz.puk.sub
-        [dns pos]:eth.sub
-        heard.etn
-        latest-block.etn
+  ++  get-source
+    |=  who=@p
+    ^-  source
+    =/  ship-source  (~(get by ship-sources.etn) who)
+    ?^  ship-source
+      (~(got by sources) u.ship-source)
+    ?:  =((clan:title who) %earl)
+      [%& (^sein:title who)]
+    (~(got by sources) default-source.etn)
+  ::
+  ++  get-source-id
+    |=  =source
+    ^-  [source-id _this-su]
+    =/  source-reverse  (~(get by sources-reverse) source)
+    ?^  source-reverse
+      [u.source-reverse this-su]
+    :-  top-source-id.etn
+    %_  this-su
+      top-source-id.etn    +(top-source-id.etn)
+      sources.etn          (~(put by sources) top-source-id.etn source)
+      sources-reverse.etn  (~(put by sources-reverse) source top-source-id.etn)
     ==
-  ::                                                    ::  ++feed:su
-  ++  feed                                              ::  subscribe to view
+  ::
+  ++  new-event
+    |=  [=a=ship =a=udiff:point]
+    ^+  this-su
+    =/  a-point=point  (~(gut by pos.zim.pki) a-ship *point)
+    =/  a-diff=(unit diff:point)  (udiff-to-diff:point a-udiff a-point)
+    ?~  a-diff
+      this-su
+    (public-keys:feel %diff a-ship u.a-diff)
+  ::
+  ++  extract-snap                                      ::  extract rewind point
```

Comment:
It looks like snapshots are no longer operational, but there are a couple stubs like this one left.  Can you explain how you see this going in the future?

Reply:
That's correct.  Now that we only listen to key changes, sponsor changes, and breaches, It seems like they're not needed, at least for now.  On mainnet it takes less than a minute of processing time to get up to date, and it doesn't block for that whole time.  It's faster on ropsten, and of course fake ships don't take any time for this.

So I've removed snapshots for now to reduce complexity.  We might decide to add them back, and we can do so much more cleanly now.